Part_2_/Person.cs

```diff
@@ -1,11 +1,68 @@
-﻿namespace Part_2
+﻿
+namespace Part_2
 {
     // TODO: inherit IPerson interface
     // TODO: create Person class with public parameters Id (long), FirstName (string), LastName (string), Email (string), Gender (Enum).
     // TODO: add method to get full name()
     // TODO: validate an email, email validation: check for NULL or Empty and contains string = "@". OnFail - thrown an ArgumentException.
     // TODO: override Equals and  GetHashCode
-    public class Person
+
+    public class Person:IPerson
     {
+        private string email;
+        public long Id { get; set; }
+        public string FirstName { get; set; }
+        public string LastName { get; set; }
+        public string Email {
+            get=>email;
+            set
+            {
+                if (string.IsNullOrEmpty(value) | !value.Contains("@")) throw new System.ArgumentException();
```

Comment:
In c# more safe to use || and &&, in case if some of conditions true next won't be executed
This logic will be wrong here